tests/mfx_dispatch/linux/mfx_dispatch_test_main.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2017-2018 Intel Corporation
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include "gtest/gtest.h"
+#include "mfx_dispatch_test_main.h"
+
+
+DlopenHook g_dlopen_hook(TEST_DLOPEN_HOOKS::AlwaysNull);
+DlsymHook g_dlsym_hook(TEST_DLSYM_HOOKS::AlwaysNull);
+FopenHook g_fopen_hook(TEST_FOPEN_HOOKS::AlwaysNull);
+MfxInitExHook g_mfxinitex_hook(TEST_MFXINITEX_HOOKS::AlwaysUnsupported);
+MfxQueryVersionHook g_mfxqueryversion_hook(TEST_MFXQUERYVERSION_HOOKS::AlwaysUnsupported);
+MfxQueryImplHook g_mfxqueryimpl_hook(TEST_MFXQUERYIMPL_HOOKS::AlwaysUnsupported);
+
+
+extern "C"
+{
+    void *dlopen(const char *filename, int flag)
+    {
+        return g_dlopen_hook(filename, flag);
+    }
+
+    void *dlsym(void *handle, const char *symbol)
+    {
+        return g_dlsym_hook(handle, symbol);
+    }
+
+    FILE * fopen(const char *filename, const char *opentype)
+    {
+        return g_fopen_hook(filename, opentype);
+    }
+
+    mfxStatus MFXInitEx(mfxInitParam par, mfxSession *session)
```

Comment:
wait... But with this you will catch MFXInitEx which application makes to the dispatcher... right? Why do you need this? I thought the idea is to return pointer to mock implementation to the dispatcher. In this way you don't need this mock implementation to be name MFXInitEx, you just need to return correct function pointer with MFXInitEx prototype when dispatcher calls dlsym with "MFXInitEx".

Reply:
Good point.